File: iteration-2/src/StudentController.java

public class StudentController {

    private Map<Integer, Student> HMstudent = new HashMap<>();//attribute studentMap as HashMap
    private List<Advisor> advisors;

    private Student getStudent(int id, String name, String surname, List<String> emails, Semester semester) {
        Student student = new Student(id, name, surname, emails, semester);
        return student;
    }

    public Map<Integer, Student> getHMstudent() {
        return HMstudent;
    }

    public void setHMstudent(Map<Integer, Student> studentMap) {
        this.HMstudent = HMstudent;
    }

    public List<Advisor> getAdvisors() {
        return advisors;
    }

    public void setAdvisors(List<Advisor> advisors) {
        this.advisors = advisors;
    }

    public void createStudent(int id, String name, String surname, List<String> emails, Semester semester) {
        //the function creates the student with given parameters such as id, name, surname, emails and semester
        Random random = new Random();
        Student student = getStudent(id, name, surname, emails, semester);
        List<Advisor> advisors = getAdvisors();
        int index = random.nextInt(advisors.size());
        Advisor advisor = advisors.get(index);
        student.setAdvisor(advisor);
        advisor.addStudent(student);
        Map<Integer, Student> HMstudent = getHMstudent();
        HMstudent.put(Integer.valueOf(id), student);
        student.setTranscript(new Transcript(student));
    }


    public void printStudents() {
        // Method to print all the Student objects in the HMstudent map
        Map<Integer, Student> HMstudent = this.HMstudent;
        // Obtain an iterator for the HMstudent map
        Iterator<Map.Entry<Integer, Student>> studentIterator = HMstudent.entrySet().iterator();
        // Iterate over the HMstudent map using the iterator
        while (studentIterator.hasNext()) {
            // Get the current entry in the map
            Map.Entry<Integer, Student> tempMap = (Map.Entry<Integer, Student>) studentIterator.next();
            // Print the student's ID and the Student object
            System.out.println(tempMap.getKey() + " = " + tempMap.getValue());
        }
    }


}

File: iteration-3/StudentController.py
import random
from Student import *

from Transcript import Transcript

class StudentController:
    def __init__(self):
        self.HMstudent = {}
        self.advisors = []

    def get_student(self, id, name, surname, emails, semester):
        student = Student(id, name, surname, emails, semester)
        return student

    def get_HMstudent(self):
        return self.HMstudent

    def set_HMstudent(self, student_map):
        self.HMstudent = student_map

    def get_advisors(self):
        return self.advisors

    def set_advisors(self, advisors):
        self.advisors = advisors

    def create_student(self, id, name, surname, emails, semester):
        student = self.get_student(id, name, surname, emails, semester)
        advisors = self.get_advisors()
        index = random.randint(0, len(advisors) - 1)
        advisor = advisors[index]
        student.set_advisor(advisor)
        advisor.add_student(student)
        HMstudent = self.get_HMstudent()
        HMstudent[id] = student
        student.set_transcript(Transcript(student))

    def print_students(self):
        HMstudent = self.HMstudent
        for key, value in HMstudent.items():
            print(key, "=", value)

StudentController.create_student()